go/pkg/client/tree.go

```diff
@@ -89,6 +110,16 @@ func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs
 			},
 		}
 		return nil
+	} else if t == command.SymlinkInputType {
+		fs[path] = &fileSysNode{
+			// TODO: How to handle a target with absolute path?
```

Comment:
[Here](https://github.com/bazelbuild/remote-apis/blob/aa8e718768c2973274c2a23e9dcc27b42afea794/build/bazel/remote/execution/v2/remote_execution.proto#L799-L801) says that an absolute path target can be supported. Do you think we still need to check if the target is within exec root or not?

Reply:
I'm fine to keep symlink as-is here, as I don't remember we have the case handling in tree absolute path symlink.
I'm fine with 2 too.

Reply:
We can error out on absolute path symlinks for now, and support them later if needed. However, traversing them is necessary. The users specify an input spec such as foo/,bar/, and it's our responsibility to decide what exactly we should upload based on that. If foo/bla is a directory symlink to /baz, we need to traverse the target to load all the files there into the Merkle tree, otherwise the symlink will be dangling on the bot.
Note that relative path symlinks that escape the input root are an error as well.